website/chatui/Program.cs

```diff
@@ -1,8 +1,29 @@
+using Microsoft.Extensions.Options;
+using chatui.Configuration;
+
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Services.AddControllersWithViews();
+builder.Services.AddOptions<ChatApiOptions>()
+    .Bind(builder.Configuration)
+    .ValidateDataAnnotations()
+    .ValidateOnStart();
+
+builder.Services.AddHttpClient("ChatGPT", (provider, client) =>
+{
+    var config = provider.GetRequiredService<IOptions<ChatApiOptions>>().Value;
+    client.BaseAddress = new Uri(config.ChatApiEndpoint);
+    client.DefaultRequestHeaders.Authorization =
+        new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", config.ChatApiKey);
+}).ConfigurePrimaryHttpMessageHandler(() =>
+{
+    return new HttpClientHandler
+    {
+        ClientCertificateOptions = ClientCertificateOption.Manual,
+        ServerCertificateCustomValidationCallback = (_, _, _, _) => true
```

Comment:
Should this only be for development? If so, can you fence this?  I know it was like this before.  (If you want, you can pick this up in PR 2 when you do end-to-end testing)

Reply:
all refinements are more than welcome for this very same PR. Please keep them coming